wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -76,47 +76,64 @@ private static function sematicDomainToLink($lang, $domain) {
 		return '<a href="' . get_site_url() . '?s=&lang=' . $lang . '&tax=' . urlencode($domain) . '">' . $domain . '</a>';
 	}
 
-	private static function entryToDisplayXhtml($id, $entry) {	
+	private static function entryToDisplayXhtml($id, $displayXhtml) {	
 		//<div class="entry" id="ge5175994-067d-44c4-addc-ca183ce782a6"><span class="mainheadword"><span lang="es"><a href="http://localhost:8000/test/ge5175994-067d-44c4-addc-ca183ce782a6">bacalaitos</a></span></span><span class="senses"><span class="sensecontent"><span class="sense" entryguid="ge5175994-067d-44c4-addc-ca183ce782a6"><span class="definitionorgloss"><span lang="en">cod fish fritters/cod croquettes</span></span><span class="semanticdomains"><span class="semanticdomain"><span class="abbreviation"><span class=""><a href="http://localhost:8000/test/?s=&amp;partialsearch=1&amp;tax=9909">1.7</a></span></span><span class="name"><span class=""><a href="http://localhost:8000/test/?s=&amp;partialsearch=1&amp;tax=9909">Puerto Rican Fritters</a></span></span></span></span></span></span></span></div></div>
-		if (isset($entry->displayXhtml) && $entry->displayXhtml !== '') {
-			$displayXhtml = Webonary_Pathway_Xhtml_Import::fix_entry_xml_links($entry->displayXhtml);
-
-			// set image and audio src path to the cloud, if they are found in the entry
-			if (preg_match_all('/src=\"(.*(?:\.jpg|.mp3))\"/iU', $entry->displayXhtml, $matches) === 1) {
-				$baseUrl = self::remoteFileUrl($entry->dictionaryId) . '/';
-				foreach($matches[0] as $index => $src) {
-					$file = str_replace("\\", "/", $matches[1][$index]);
-					$displayXhtml = str_replace($src, 'src="' . $baseUrl . $file . '"', $displayXhtml);
-				}
+		$displayXhtml = Webonary_Pathway_Xhtml_Import::fix_entry_xml_links($displayXhtml);
+
+		// set image and audio src path to the cloud, if they are found in the entry
+		if (preg_match_all('/src=\"(.*(?:\.jpg|.mp3))\"/iU', $displayXhtml, $matches) > 0) {
+			$baseUrl = self::remoteFileUrl($entry->dictionaryId) . '/';
```

Comment:
`$entry` is undefined, probably should be `$displayXhtml`.

Reply:
Yes, we do need unit tests for all the new cloud api code. I've been hesitant to put them in, because we were still trying to figure out how all this end up working, and coordinating changes with FLex. Now that things are more stable, I have created an issue to make sure to put in unit tests. https://github.com/sillsdev/webonary/issues/72